src/core_editor/editor.rs

```diff
@@ -0,0 +1,189 @@
+use super::{Clipboard, LineBuffer};
+
+pub struct Editor {
+    // Hack: Fix this once coupling in the code is taken care of
+    pub line_buffer: LineBuffer,
```

Comment:
As mentioned in the caveat section: I don't really think this should be public at all, or that engine should even know if there is a line buffer under the hood (editor).

Reply:
I would think of `LineBuffer` as the representation that is allowed to be `Clone` and `Send` whilst the engine with its undo capabilities would get out of step if that would be the case. Adding a setter and getter is certainly an option (Setting the line buffer to something new from outside (plugin etc.) would also have implications for the undo behavior)

Reply:
I intended to pull undo/redo as editor specific features as you can see in #129 . Given all these pieces are encapsulated inside the editor. I don't really think there is the scope of getting anything get out of step.

Reply:
A completion/suggestion might want to set the content of the line buffer to a new value, including a useful cursor position. In this case supplying a `LineBuffer` instance might be an easy api. Yet the undo stack managed by `Editor` would need to keep track that the linebuffer changed in a certain way to undo this completion. So a setter might be a better choice than the `pub line_buffer`